climada/hazard/low_flow.py

```diff
@@ -571,39 +708,120 @@ def _read_single_nc(filename, yearrange, bbox):
             data = data.sel(time=slice(cftime.DatetimeNoLeap(yearrange[0], 1, 1),
                                        cftime.DatetimeNoLeap(yearrange[-1], 12, 31)))
         else:
-            data = data.sel(lon=slice(bbox[0], bbox[2]), lat=slice(bbox[3], bbox[1]),
+            lon_min, lat_min, lon_max, lat_max = bbox
+            data = data.sel(lon=slice(lon_min, lon_max), lat=slice(lat_max, lat_min),
                             time=slice(cftime.DatetimeNoLeap(yearrange[0], 1, 1),
                                        cftime.DatetimeNoLeap(yearrange[-1], 12, 31)))
         datetimeindex = data.indexes['time'].to_datetimeindex()
         data['time'] = datetimeindex
     return data
 
 
+def _xarray_reduce(data, fun=None, percentile=None):
+    """wrapper function to reduce xarray along time axis
+
+    Parameters:
+        data (xarray)
+
+    Optional Parameters:
+        fun (str): function to be applied, either "mean" or "percentile"
+        percentile (num): percentile to be extracted, e.g. 5 for 5th percentile
+            (only if fun=='percentile')
+
+    Returns:
+        xarray
+    """
+    if fun == 'mean':
+        return data.mean(dim='time')
+    if fun[0] == 'p':
+        return data.reduce(np.nanpercentile, dim='time', q=percentile)
+    return None
+
+def _split_bbox(bbox, width=BBOX_WIDTH):
+    """split bounding box into squares, return new set of bounding boxes
```

Comment:
Could this function be a candidate for climada.util in the future?

Reply:
yes, but i will not ouch climada.util for this release

Reply:
i will add a comment in the code